wallet-new/src/Cardano/Wallet/API/Request/Parameters.hs

```diff
@@ -0,0 +1,11 @@
+-- | Enlists names of request parameters.
+-- To import only in /qualified/ way.
+
+module Cardano.Wallet.API.Request.Parameters where
+
+type WalletId = "wallet_id"
+type TransactionId = "id"  -- it's quite unambigous for transactions
+type CreatedAt = "created_at"
+type Balance = "balance"
```

Comment:
To clarify what I meant by ambiguity (to not come across as a grumpy man which changes his ideas every other day 😁 ):

As this is a REST API and we are manipulating resources, my rule of thumb would be:

- If we are filtering over the manipulated resource, we use simply `id`. What I mean by "manipulated resource"? If we are querying `/api/wallets` or `api/transactions` it's clear that the manipulated resource is a `Wallet` is the first case and a `Transaction` in the latter, therefore for both we should have just "id" (and yes, on line 6 we should change `wallet_id` to simply wallet, me thinks.

This way our filter will be nice and clean `/api/v1/wallets?id=BlaBla` and `api/v1/transactions?id=1ad2...`

- If there is ambiguity because we are filtering on a resource related to the one manipulated, than we need to qualify things. For example, let's assume the fictional case where we want to filter the wallet list by transaction ids, we would write `/api/v1/wallets?transaction_id=blabla`.

I was initially concerned (by self-reviewing my code) that having `id` being used in more than 1 place would have created conflicts but I think it should be fine, as we have the following type family:

```
type family FilterParams (syms :: [Symbol]) (r :: *) :: [*] where
    FilterParams '["wallet_id", "balance"] Wallet = IndicesOf Wallet
```

This should ensure, thanks to the fact it's parametrised over `r` that things wouldn't conflicts:

```
type family FilterParams (syms :: [Symbol]) (r :: *) :: [*] where
    FilterParams '["id", "balance"] Wallet = IndicesOf Wallet
    FilterParams '["id", "created_at"] Transaction = IndicesOf Transaction
```

@Martoon-00 , @parsonsmatt thoughts?

Reply:
Looks very sane for me.